libcnb/examples/example-02-ruby-sample/src/layers/bundler.rs

```diff
@@ -56,50 +64,38 @@ impl LayerLifecycle<RubyBuildpack, BundlerLayerMetadata, HashMap<String, String>
             .wait()?;
 
         if !bundle_exit_code.success() {
-            return Err(anyhow::anyhow!("Could not install gems!"));
+            return Err(anyhow::anyhow!("Could not bundle install"));
         }
 
-        Ok(LayerContentMetadata {
-            types: LayerTypes {
-                build: false,
-                launch: true,
-                cache: true,
-            },
-            metadata: BundlerLayerMetadata {
-                gemfile_lock_checksum: sha256_checksum(build_context.app_dir.join("Gemfile.lock"))?,
-            },
+        LayerResultBuilder::new(BundlerLayerMetadata {
+            gemfile_lock_checksum: sha256_checksum(context.app_dir.join("Gemfile.lock"))?,
         })
+        .build()
     }
 
-    fn validate(
+    fn existing_layer_strategy(
```

Comment:
I had a list of naming ideas here, if it helps for inspiration:
https://github.com/Malax/libcnb.rs/pull/144#discussion_r751338701

I was originally going to include "handle layer" in that list of ideas, however to me, "handle" means actually do something with the layer, whereas this function is supposed to just determine the action to take, rather than actually take it.

Reply:
I like the naming as-is (Thanks again @edmorley!), I see it as a getter (which don't use the `get_` prefix in Rust by convention) that allows libcnb to get the existing layer strategy which is exactly what this is. This is reinforced through the return type: `ExistingLayerStrategy`.

I think `handle` would be very misleading as this method does't handle the existing layer strategy, it merely returns one and libcnb does the handling.